site/learn/expert-advice/security/ssl-vs-tls.md

```diff
@@ -0,0 +1,51 @@
+
+Now, the overall process of this handshake protocol looks like this:
+
+
+1. The client sends a “clienthello” message to the server. This message includes information like SSL/TLS version, and the cryptographic algorithm and data compression methods that the client can support.  
+2. The server responds with a “serverhello” message. This message includes cryptographic algorithms that the server has chosen from the list provided by the client. In this way, an algorithm that both sides support is chosen. The server also sends a session Id, its digital certificate, and its public key.
+3. The client verifies the validity of the server’s certificate with the entity that issued it. It does so by first checking to ensure that the server's certificate is not expired and the domain name or IP address on the certificate matches the server's information. Then, the client attempts to verify that the server's certificate has been signed by the certificate authority who authorized it. It does so by verifying that the signature on the certificate has been signed by the certificate authority's public key.
+4. A secret key is exchanged. The client sends a shared secret key to be used between the client and the server. The secret key is encrypted with the server’s public key.
+5. If required, the server verifies the client's certificate.
+6. The client sends a “finished” message to the server, which is encrypted with the secret key. This confirms that the client section of the handshake protocol is completed.
+7. The server responds with a “finished” message, encrypted with the secret key. This indicates the server part of the handshake protocol is finished.  
+8. The client and server now exchange messages that are symmetrically encrypted with the shared secret key.
+
+
+{% plantuml source: _diagrams/learn/expert-advice/ssl-vs-tls/ssl-handshake.plantuml, alt: "The SSL/TLS handshake protocol." %}
+
+Tl;dr, the entire purpose of the handshake is to verify the validity of the server. The handshake also securely exchanges a shared secret key, which is used to encrypt and decrypt messages sent between the client and the server.
+
+*Why should I care?*
+
+Imagine you are sending a very important message to someone through the mail, that if it was intercepted and read along the way, would have disastrous consequences for your life.  You would need some way of coding the message so if someone finds it, they can’t understand it (encryption) and putting it in an envelope so that the message is secure (securing the communication).  In the same way, information sent over the internet is assumed to need encryption and security, because all kinds of data needs to be sent over the web: credit card numbers, secret love notes, health information, you name it. It all needs to be encrypted and secured, and the way that happens currently is with SSL/TLS.
```

Comment:
Does it make sense to have this be at the start? "why should you care" is usually a great way to start a post.